pos_voucher/static/src/js/screens.js

```diff
@@ -191,6 +192,52 @@ var voucherListScreenWidget = screens.ScreenWidget.extend({
 });
 gui.define_screen({name:'voucherlist', widget: voucherListScreenWidget});
 
+
+var popup_voucher_generate = PopupWidget.extend({
+        template: 'popup_voucher_generate',
+        init: function (parent, options) {
+            this._super(parent, options);
+        },
+        show: function (options) {
+            var self = this;
+            this._super(options);
+            this.$(".cancel").click(function (e) {
+                self.pos.gui.close_popup();
+            });
+            this.$('.input_form').focus();
+            this.list = options.list || [];
+        },
+        renderElement: function () {
+            var self = this;
+            this._super();
+            this.$('.cancel').click(function () {
+                self.gui.close_popup();
+            });
+            this.$('.confirm').click(function () {
+                var voucher_amount = $('#voucher_amount').val();
+                var voucher_type_selection = $('#voucher_type_selection').val();
+                var order    = self.pos.get_order();
+                if (voucher_amount > 0 && voucher_type_selection){
+                    var select_voucher = self.pos.voucher_type_by_id[voucher_type_selection];
+                    if (select_voucher){
+                        var product  = self.pos.db.get_product_by_id(select_voucher['product_id'][0]);
+                        if (product){
+                            order.add_product(product, {
+                                price: voucher_amount,
+                                extras: { voucher_type_id: select_voucher.id },
+                            });
+                        }
+                    }
+                }
+            })
+        }
+    });
+    gui.define_popup({
+        name: 'popup_voucher_generate',
+        widget: popup_voucher_generate
+    });
+
+
 return {
     voucherListScreenWidget: voucherListScreenWidget,
```

Comment:
```suggestion
    voucherListScreenWidget: voucherListScreenWidget,
    popupVoucherGenerate: popupVoucherGenerate,
```

We want to return this object so that future modules can extend it